akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -163,6 +168,17 @@ private[cluster] class ClusterRemoteWatcher(
     }
   }
 
+  override def addWatch(watchee: InternalActorRef, watcher: InternalActorRef): Unit = {
+    val watcheeNode = watchee.path.address
+    if (!clusterNodes.contains(watcheeNode) && memberTombstones.exists(_.address == watcheeNode)) {
```

Comment:
Thinking about if the `memberTombstones` should be converted to `Set[Address]` when changed to avoid the exists iteration here.
It's probably rare to try to watch something that's not in the cluster any more so I think this is fine, and better to not convert when changed.